src/cli/Peers.ts

```diff
@@ -0,0 +1,387 @@
+import fs from 'fs';
+import commander from 'commander';
+import { PolykeyAgent, PeerInfo } from '../lib/Polykey';
+import {} from '../../tmp/hole-punch-test/publicserver';
+import { actionRunner, pkLogger, PKMessageType, determineNodePath } from '.';
+
+function makeAddPeerCommand() {
+  return new commander.Command('add')
+    .description('add a new peer to the store')
+    .option('-b, --base64 <base64>', 'decode the peer info from a base64 string')
+    .option('-p, --public-key <publicKey>', 'path to the file which contains the public key')
+    .option('-a, --connected-address <connectedAddress>', 'public address on which the node can be contacted')
+    .option('-r, --relay-key <relayKey>', 'path to the file which contains the public key of the relay peer')
+    .option('--node-path <nodePath>', 'node path')
+    .option('-v, --verbose', 'increase verbosity level by one')
+    .action(
+      actionRunner(async (options) => {
+        const client = PolykeyAgent.connectToAgent();
+        const status = await client.getAgentStatus();
+        if (status != 'online') {
+          throw Error(`agent status is: ${status}`);
+        }
+
+        const nodePath = determineNodePath(options);
+        const base64String = options?.base64?.replace('\r', '')?.replace('\n', '');
+
+        let successful: boolean;
+        if (base64String != undefined) {
+          // read in peer info string
+          const { publicKey, connectedAddress, relayPublicKey } = PeerInfo.parseB64(base64String);
+          successful = await client.addPeer(nodePath, publicKey, connectedAddress?.toString() ?? '', relayPublicKey);
+        } else {
+          const publicKeyPath = options.publicKey;
+          // read in publicKey if it exists
+          const publicKey = fs.readFileSync(publicKeyPath).toString();
+
+          const connectedAddress = options.connectedAddress;
+          const relayPublicKeyPath = options.relayKey;
+          const relayPublicKey = fs.readFileSync(relayPublicKeyPath).toString();
+
+          successful = await client.addPeer(nodePath, publicKey, connectedAddress, relayPublicKey);
+        }
+
+        if (successful) {
+          pkLogger('peer successfully added to peer store', PKMessageType.SUCCESS);
+        } else {
+          pkLogger('something went wrong, peer was not added to peer store', PKMessageType.WARNING);
```

Comment:
Something I learned from the Python logging system, is to carry the "module" in the logger itself.

So then the idea is that the logger is instantiated from a logger constructor. When you do so you are then creating the logger "module" as well, that is a key that indicates where the logging came from.

So we already have a form of separation in the sense of "log levels". I see that. But then with the "key", it's a like trace to represent hierarchical logging. Can then be useful to group logs by the keys and show how the whole thing is nested.

This is also perfect for tracing, which is why I'd like to explore the opentracing capabilities for js-polykey too and asynchronous debugging!